fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/service/LoanChargeWritePlatformServiceImpl.java

```diff
@@ -1041,7 +1041,8 @@ private boolean addCharge(final Loan loan, final Charge chargeDefinition, LoanCh
 
         // we want to apply charge transactions only for those loans charges that are applied when a loan is active and
         // the loan product uses Upfront Accruals
-        if (loan.getStatus().isActive() && loan.isNoneOrCashOrUpfrontAccrualAccountingEnabledOnLoanProduct()) {
+        if ((loan.getStatus().isActive() && loan.isNoneOrCashOrUpfrontAccrualAccountingEnabledOnLoanProduct())
+                || (loan.getStatus().isOverpaid() && loan.isPeriodicAccrualAccountingEnabledOnLoanProduct())) {
```

Comment:
Would you mind to change the condition to:
- Allow to create the accrual transaction if the loan is closed? I dont think we need the "is periodic accrual enabled" and also CLOSED_OBLIGATIONS_MET is a valid outcome after the charge was added:
- It was overpaid but charge was added and now it is closed (edge case, but why not...)

Reply:
Done! conditions added!